### PR TITLE
Use Python prefix for include and library dirs

### PR DESCRIPTION
**Pull request by jakirkham**
Needed to fix some compilation issues with the Conda package.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,14 +12,16 @@
...
 
+import os
+from distutils.sysconfig import get_config_var, get_python_inc
 from setuptools import setup, Extension
 
 module = Extension(
     'ptxcompiler._ptxcompilerlib',
     sources=['ptxcompiler/_ptxcompilerlib.cpp'],
-    include_dirs=['/usr/local/cuda-11.5/include'],
+    include_dirs=[os.path.dirname(get_python_inc())],
```
Comment (leofang) [inline setup.py:22]: Don't we also need CUDA includes?

File: setup.py [hunk 2]
```diff
@@ -12,14 +12,16 @@
...
     libraries=['nvptxcompiler_static'],
-    library_dirs=['/usr/local/cuda-11.5/lib64'],
+    library_dirs = [get_config_var("LIBDIR")],
```
Comment (leofang) [inline setup.py:24]: ditto

File: setup.py (hunk 1 above)
Comment (jakirkham) [reply setup.py:22]: Nope

https://github.com/conda-forge/nvcc-feedstock/blob/295b65a0c67be9d81e7f4812cd495a9e95dcae57/recipe/install_nvcc.sh#L73-L75

File: setup.py (hunk 2 above)
Comment (jakirkham) [reply setup.py:24]: This is actually where the libraries with the exception of those in the sysroot. We symlink [`libcuda.so`]( https://github.com/conda-forge/nvcc-feedstock/blob/295b65a0c67be9d81e7f4812cd495a9e95dcae57/recipe/install_nvcc.sh#L105 ) there. We then symlink the remaining one in the recipe

Comment (leofang) [reply setup.py:24]: Sounds good, thanks for reminders John!

**Comment by jakirkham** [comment]
That said, if it is helpful for other use cases, we could keep `/usr/local/cuda` around and rely on that pointing to the right CUDA version. We could also check the `CUDA_HOME` environment variable, which is set in conda-forge, but a user could also override as well. Just some other thoughts 🙂

**Comment by gmarkall** [comment]
If we use the Python include and lib dirs, do we also need a dependency on a cudatoolkit package (that includes the `libptxcompiler_static.a`) to ensure that the library and headers are available?

**Comment by jakirkham** [comment]
Yeah this is what Leo was asking about above.

Basically conda-forge has an `nvcc` shim package, which does a few things (like making sure the right headers are picked up). With the static library, `libnvptxcompiler_static.a`, we symlinked that into [the right location]( https://github.com/conda-forge/staged-recipes/pull/16929/files#diff-dc1ea741197faebecbd5f57749c7b6d4d9cf8ce143ea60f792fc0501d917aa15R13 ) since it isn't in the `cudatoolkit` package. As to the `cudatoolkit` package itself, this gets added [as a dependency during the build]( https://github.com/conda-forge/nvcc-feedstock/blob/295b65a0c67be9d81e7f4812cd495a9e95dcae57/recipe/meta.yaml#L34-L40 ).

**Comment by gmarkall** [comment]
At the moment though the recipe makes no mention of any `cudatoolkit` or `nvcc` packages, because I relied on it being installed on my local system: https://github.com/rapidsai/ptxcompiler/blob/main/conda/recipes/ptxcompiler/meta.yaml - does a `cudatoolkit` or `nvcc` entry need adding to `meta.yaml` as well?

**Comment by jakirkham** [comment]
We add [`{{ compiler("cuda") }}` to the conda-forge recipe]( https://github.com/conda-forge/staged-recipes/pull/16929/files#diff-dc1ea741197faebecbd5f57749c7b6d4d9cf8ce143ea60f792fc0501d917aa15R20 ). This references the `nvcc` shim package currently.

Not sure if I would worry about that outside of conda-forge. It does some conda-forge specific things like [symlinking the `libcuda.so` stub in place]( https://github.com/conda-forge/nvcc-feedstock/blob/295b65a0c67be9d81e7f4812cd495a9e95dcae57/recipe/install_nvcc.sh#L105 ). That makes sense in the Docker image conda-forge builds in, but not sure if that makes sense outside of that environment.

I guess that's a longwinded of saying no I think we can skip that here. We can revisit in the future as things change in the Conda ecosystem.

**Pull request merged**